packages/lib-classifier/src/store/RootStore.js

```diff
@@ -83,12 +83,28 @@ const RootStore = types
       self.subjects.reset()
     }
 
+    function _afterInitialize() {
+      // Because the root store is initialized all the sub-stores are as well
+      // We enable listening between this root store's stores by calling an afterRootInitialize() method on all these sub-stores
+      const allKeys = keys(self);
+      allKeys.forEach(key => {
+        // Filter the current model to only the observable properties of the root store since that's all our models
+        const isFunction = typeof self[key] === 'function';
+        const isVolatile = !isObservableProp(self, key);
+        if (isFunction || isVolatile) return;
+
+        // If the sub-store has the afterRootInitialize() method, run it
+        if (self[key].afterRootInitialize) self[key].afterRootInitialize();
+      });
+    }
```

Comment:
The UI is looking good for this PR! A clarifying question though in order to complete my review:

it looks like the only purpose of this `_afterInitialize()` function is to run `afterRootInitialize()` specifically on the SubjectViewerStore. Why did you decide to create `_afterInitialize()` in the RootStore and loop through all of its child stores? Why not just call `afterRootInitialize()` in SubjectViewerStore's `afterCreate()` or `afterAttach()` method? (MST [lifecycle method](https://mobx-state-tree.js.org/overview/hooks) docs)

Reply:
To further why I'm asking - In the `afterRootInitialize()` function you've added to SubjectViewerStore, it's using a `reaction` with the note "reactions handle any changes to state". In the scenario where SubjectViewerStore gets created prior to WorkflowStepsStore, once WorkflowStepsStore _does_ get created, doesn't that mean `getRoot(self)?.workflowSteps.hasAnnotateTask` changes from `false` to `true`? And the `reaction` should catch that change in state?

I want to be sure that using `mobx` methods instead of `mobx-state-tree` methods in this scenario is not undermining the purpose of using `mobx-state-tree` for React.

Reply:
@goplayoutside3 - the reason for this is because of how Javascript itself works and the execution order of the code. The root store is initialized with either an existing sub-model - `ClassificationStore` as an example - or creates one itself for reference when the root store is created... But this is what causes the problem. `afterCreate()` and `afterAttach()` run at the end of the model creation... Which means that all the sub-models `afterCreate()` and `afterAttach()` are called before the `ClassifierRootStore.afterCreate()` and `ClassifierRootStore.afterAttach()` are called... What this does is cause `getRoot(self)?.workflowSteps` to be _undefined_ in `WorlflowStepsStore.afterCreate()` and `SubjectViewerStore.afterCreate()` because the `RootStore` itself isn't finished with setup - the sub models need to finish setup before it itself finishes setup. Binding on an undefined `RootStore` causes this feature to not work, whereas when the data in the models get refreshed - i.e. they're already existent but updated - then the lifecycle hooks work properly.

So, the insight I had to make the references existent (and therefore watchable/listenable/connectable on initial load) is to create an `afterCreate()` hook around the `RootStore.afterCreate()` that invokes a method on each of sub-models when the context of the `RootStore` model has been fully setup and is no longer undefined/null. It's essentially a recursion on the aspect-oriented programming concept of `afterCreate()` applied to the sub-stores of the RootStore(). Essentially, `RootStore.afterCreate()` invokes `RootStoreSubModel.afterRootCreate()` after RootStore() referentially exists.

My hunch is that this new lifecycle method will be useful in avoiding many of the current edge cases in inter-relational-dependencies between peer sub-models in the RootStore.